compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -49,25 +49,34 @@ object ProtoTypes {
     /** Test compatibility after normalization.
      *  Do this in a fresh typerstate unless `keepConstraint` is true.
```

Comment:
That comment doesn't exactly reflect the subtle semantics of `keepConstraint`, perhaps we can simply delete that line since the details are documented in the implementation anyway, or say something like:
```suggestion
     * If `keepConstraint` is false, the current constraint set will not be modified by this call.
```